ipaserver/install/bindinstance.py

```diff
@@ -423,17 +423,15 @@ def zonemgr_callback(option, opt_str, value, parser):
             encoding = getattr(sys.stdin, 'encoding', None)
             if encoding is None:
                 encoding = 'utf-8'
-            value = value.decode(encoding)
+            if not isinstance(value, unicode):
```

Comment:
I had some more time and I see now why you did this here. `OptionParser` behaves differently in Python 2/3 so it actually makes sense to do this. If you could probably drop a comment there.
`value` will never be of `bytes` type, though, I believe you did not express yourself well in the commit message. In Python 3, `value` is of `str` type which does not have `decode()`. Unfortunately, it's also of `str` type in Python 2 which does has the `decode()` method.